ludwig/config_validation/checks.py

```diff
@@ -322,6 +341,45 @@ def check_tagger_decoder_requirements(config: "ModelConfig") -> None:  # noqa: F
         )
 
 
+@register_config_check
+def check_hyperopt_parameter_dicts(config: "ModelConfig") -> None:  # noqa: F821
+    """Checks for hyperopt parameter dicts against their config objects."""
+    from ludwig.schema.hyperopt.utils import get_parameter_cls, parameter_config_registry
+
+    if config.hyperopt is not None:
```

Comment:
nit: Use less nesting for better code readability.

```python
if config.hyperopt is None:
    return
<rest of code.>
```

Reply:
What do you think about adding something like a `hyperopt_check` decorator that includes the `if config.hyperopt is not None` logic? That would set us up nicely to encapsulate any future constraints and make the current checks a bit cleaner.

Reply:
I think that's a reasonable idea, though one I would suggest doing outside of this PR. There are other potentially reusable aux check preconditions (`if ecd`, `if gbm`, etc.) that would be good to consider in the design of how to extend the existing decorator.